client/starwhale/api/_impl/job.py

```diff
@@ -1,30 +1,39 @@
 import typing as t
+import numbers
 from pathlib import Path
 
 import yaml
 from loguru import logger
 
-from starwhale.consts import DEFAULT_EVALUATION_JOB_NAME, DEFAULT_EVALUATION_RESOURCE
+from starwhale.consts import DEFAULT_EVALUATION_JOB_NAME
 from starwhale.core.job import dag
 from starwhale.utils.fs import ensure_file
 from starwhale.utils.load import load_module
 
+resource_names = ["request", "limit"]
+
 
 def step(
     job_name: str = DEFAULT_EVALUATION_JOB_NAME,
-    resources: t.Optional[t.List[str]] = None,
+    resources: t.Optional[t.Dict[str, t.Any]] = None,
     concurrency: int = 1,
     task_num: int = 1,
     needs: t.Optional[t.List[str]] = None,
 ) -> t.Any:
-    _resources = resources or [
-        DEFAULT_EVALUATION_RESOURCE,
-    ]
+    _resources = resources or {}
     _needs = needs or []
 
     def decorator(func: t.Any) -> t.Any:
         if Parser.is_parse_stage():
-            cls, delim, func_name = func.__qualname__.rpartition(".")
+            cls, _, func_name = func.__qualname__.rpartition(".")
+            for _name, _resource in _resources.items():
+                if isinstance(_resource, numbers.Number):
+                    _resources[_name] = {"request": _resource, "limit": _resource}
+                elif isinstance(_resource, dict):
+                    if not all(n in resource_names for n, _ in _resource.items()):
```

Comment:
`n in resource_names for n, _ in _resource.items()` --> `n in resource_names for n in _resource` ?

Reply:
done